tasks/42_first-n-smallest/first-n-smallest.js

```diff
@@ -0,0 +1,18 @@
+export const firstNSmallest = (array, n) => {
+  let a = array.slice();
+  let small = [];
+  let result = [];
+  a.sort((a, b) => a - b);
+
+  for (let i = 0; i < a.length && i < n; i++) {
+    small.push(a[i]);
+  }
+
+  for (let i = 0; i < array.length; i++) {
+    if (small.includes(array[i])) {
+      result.push(array[i]);
+      small.splice(small.indexOf(array[i]), 1);
+    }
+  }
+  return result;
```

Comment:
Do we really need this many iterations through the array? Is there a more efficient way?
Hint: can we somehow sort the array but keep the info about original index of each element?